src/Input/Input.tsx

```diff
@@ -22,10 +26,10 @@ export interface InputProps {
  * in this component and should be handled in the consuming app.
  *
  */
-const Input: FC<InputProps & ComponentPropsWithRef<'input'>> = forwardRef(({ label, error = '', ...props }, ref) => (
+const Input: FC<InputProps & ComponentPropsWithRef<'input'>> = forwardRef(({ label, error = '',required, ...props }, ref) => (
         <ThemeProvider theme={theme}>
             <StyledLabel error={error}>
-                {label && <span>{label}</span>}
+                {label && <span>{label} {required && <span className="required-field">*</span>}</span>}
```

Comment:
Rather than have the empty white-space exist everytime (whether there is an asterisk or not), it should be part of the required logic (i.e. somewhere after the `&&`, before or inside the span).